biz.aQute.bndlib/src/aQute/bnd/component/DSAnnotations.java

```diff
@@ -35,10 +35,12 @@
 		inherit, felixExtensions, extender
 	};
 
+	private static final Parameters	DEFAULT_HEADER	= OSGiHeader.parseHeader("*");
+
 	public boolean analyzeJar(Analyzer analyzer) throws Exception {
 		Parameters header = OSGiHeader.parseHeader(analyzer.getProperty(Constants.DSANNOTATIONS));
```

Comment:
Instead of the DEFAULT_HEADER, do `analyzer.getProperty(Constants.DSANNOTATIONS, "*")`. 

header.size()==0 will match `-dsannotations: `.